compiler/src/dotty/tools/dotc/transform/localopt/Optimisation.scala

```diff
@@ -0,0 +1,20 @@
+package dotty.tools.dotc
+package transform.localopt
+
+import core.Contexts.Context
+import ast.tpd.Tree
+
+trait Optimisation {
+
+  /** Run first to gather information on Trees (using mutation) */
+  def visitor: Tree => Unit
+
+  /** Does the actual Tree => Tree transformation, possibly using a different
+   *  context from the one using in Optimisation.
```

Comment:
Could you give more explanations on what contexts are used when? Also typo: using -> used.